crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -135,11 +137,15 @@ impl Extension for QueryLimitsChecker {
         variables: &Variables,
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
+        let instant = Instant::now();
         let cfg = ctx
             .data::<ServiceConfig>()
             .expect("No service config provided in schema data");
 
         if query.len() > cfg.limits.max_query_payload_size as usize {
+            if let Some(metrics) = ctx.data_opt::<Arc<Metrics>>() {
```

Comment:
Should we also observe the query length in this case?

Reply:
Why not?

Reply:
Added!